utils/vptr/include/virtual_ptr.hpp

```diff
@@ -228,10 +230,15 @@ class PointerMapper {
   }
 
   /* get_buffer.
-   * Returns a buffer from the map using the pointer address
+   * Returns a pointer to a buffer from the map using the pointer address
    */
-  buffer_t get_buffer(const virtual_pointer_t ptr) {
-    return get_node(ptr)->second._b;
+  template <typename buffer_allocator = default_buffer_allocator>
+  cl::sycl::buffer<buffer_data_type, 1, buffer_allocator> get_buffer(
+      const virtual_pointer_t ptr) {
+    using buffer_t = cl::sycl::buffer<buffer_data_type, 1, buffer_allocator>;
+    auto buf = &get_node(ptr)->second._b;
+    buffer_t *b = static_cast<buffer_t *>(buf);
```

Comment:
Also, I don't think you need to pass the allocator type to the get_buffer if you use the typed allocator from the allocator headers of computecpp.